src/pages/task/actionButtons/PreviousCommits.tsx

```diff
@@ -0,0 +1,212 @@
+  useEffect(() => {
+    if (versionMetadata?.isPatch === false) {
+      fetchParentTask({
+        variables: {
+          projectIdentifier,
+          skipOrderNumber,
+          buildVariantOptions: {
+            ...bvOptionsBase,
+          },
+        },
+      });
+    } else if (versionMetadata?.isPatch === true) {
+      setParentTask(baseTask);
+    }
+  }, [versionMetadata]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Hook to find last passing & last executed tasks. This is a separate hook because the parent task must be
+  // known before proceeding.
+  useEffect(() => {
+    if (parentTask) {
+      if (parentTask.status !== TaskStatus.Succeeded) {
+        fetchLastPassing({
+          variables: {
+            projectIdentifier,
+            skipOrderNumber,
+            buildVariantOptions: {
+              ...bvOptionsBase,
+              statuses: [TaskStatus.Succeeded],
+            },
+          },
+        });
+      }
+      if (!isFinishedTaskStatus(parentTask.status)) {
+        fetchLastExecuted({
+          variables: {
+            projectIdentifier,
+            skipOrderNumber,
+            buildVariantOptions: {
+              ...bvOptionsBase,
+              statuses: finishedTaskStatuses,
+            },
+          },
+        });
+      }
+    }
+  }, [parentTask]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // The default link should be undefined so that the GO button is disabled if links do not exist.
+  let link: string;
+  if (parentTask) {
+    switch (selectState) {
+      case CommitType.Base:
+        link = getTaskRoute(parentTask.id);
+        break;
+      // If a base task succeeded, the last passing commit is the base task.
+      case CommitType.LastPassing:
+        link = getTaskRoute(lastPassingTask?.id || parentTask.id);
+        break;
+      // If a base task exists and has finished, the last executed commit is the base task.
+      case CommitType.LastExecuted:
+        link = getTaskRoute(lastExecutedTask?.id || parentTask.id);
+        break;
+      default:
+    }
+  }
+
+  return versionMetadata?.isPatch !== undefined ? (
+    <PreviousCommitsWrapper>
+      <StyledSelect
+        size="small"
+        data-cy="previous-commits"
+        label="Previous commits for this task"
+        allowDeselect={false}
+        onChange={(v: CommitType) => setSelectState(v)}
+        value={selectState}
+        disabled={!versionMetadata?.baseVersion}
+      >
+        <Option value={CommitType.Base} key={CommitType.Base}>
+          Go to {versionMetadata?.isPatch ? "base commit" : "parent commit"}
+        </Option>
+        <Option value={CommitType.LastPassing} key={CommitType.LastPassing}>
+          Go to last passing version
+        </Option>
+        <Option value={CommitType.LastExecuted} key={CommitType.LastExecuted}>
+          Go to last executed version
+        </Option>
```

Comment:
What happens if the task never passed or its the first time it ever ran?

Reply:
Hmm so I checked using this project `http://localhost:3000/commits/docs-ecosystem?skipOrderNumber=8`

I'm pretty sure what happens is that all the links redirect to the parent task. Which is slightly unexpected behavior, because if it never passed it'll link to the parent task (which didn't pass) and if it didn't finish executing it'll link to the parent task (which hasn't finished executing). Maybe I can disable the `Go` button in these cases?